### DISCUSSION TITLE
Logging to a file - Formatting

### DISCUSSION THREAD

**Discussion by Maxximiliann** [community/community#21462] [Other Feature Feedback, Questions, & Ideas]
<pre><code class="lang-auto">#logging.conf

[loggers]
keys=root,activity_report

[handlers]
keys=fileHandler

[formatters]
keys=activity_report_format

[logger_root]
level=INFO
handlers=fileHandler
formatter=activity_report_format

[logger_activity_report]
level=INFO
handlers=fileHandler
formatter=activity_report_format
qualname=activity_report
propogate=0

[handler_fileHandler]
class=handlers.RotatingFileHandler
level=INFO
fomatter=activity_report_format
args=('activity_report.txt', 26214400, 5) #25mb = 26,214,400 bytes

[formatter_activity_report_format]
format=%(asctime)s - %(name)s - %(levelname)s - %(message)s
class=logging.Formatter
</code></pre>
<pre><code class="lang-auto">#Sandbox.py

import logging
import logging.config
from logging.handlers import RotatingFileHandler

logging.config.fileConfig('/home/maxximiliann/Desktop/Arbit/lib/Python/logging.conf')
logger = logging.getLogger(__name__)

logger.error('Trouble in Paradise . . . ')
</code></pre>
<pre><code class="lang-auto">
#activity_report.txt

Trouble in Paradise . . .
</code></pre>
<p>What more needs to be done such that logs are recorded to <code>activity_report.txt</code> with the formatting defined in <code>activity_report_format</code> ?</p>

**Comment by fire-eggs** [comment]
<ol>
<li>“propagate” is misspelled.</li>
<li>By using <code>__name__</code> to get the logger, you’re requesting (e.g) “sandbox.py” or something. You want to do <code>logger = logging.getLogger("activity_report")</code> or something like that to get your desired logger. [or <code>getLogger('')</code> to get the root logger.</li>
</ol>

**Comment by Maxximiliann** [comment, marked as answer]
<p>Recently discovered <a href="https://loguru.readthedocs.io/en/stable/" rel="noopener nofollow ugc">Loguru</a>. Definitely a much better way to do logging but thanks for your help :slight_smile:</p>
